### PR TITLE
fix broken debian builds. debian 7 should be used if debian 7 dotdeb …

### LINKED ISSUE #433: Tests failing under debian8
in multiple PR the tests fail with debian8

- https://github.com/voxpupuli/puppet-php/pull/432 https://travis-ci.org/voxpupuli/puppet-php/jobs/355132633
- https://github.com/voxpupuli/puppet-php/pull/431 https://travis-ci.org/voxpupuli/puppet-php/jobs/354785257
- https://github.com/voxpupuli/puppet-php/pull/292  https://travis-ci.org/voxpupuli/puppet-php/jobs/355585742

```
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
php5-cli : Depends: libdb5.1 but it is not installable
          Recommends: php5-readline but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
Error: /Stage[main]/Php::Packages/Package[php5-cli]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
php5-cli : Depends: libdb5.1 but it is not installable
          Recommends: php5-readline but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

- in debian7 wheezy https://packages.debian.org/wheezy/php5-cli is build against https://packages.debian.org/wheezy/libdb5.1
- in debian8 jessie https://packages.debian.org/jessie/php5-cli against https://packages.debian.org/jessie/libdb5.3
- in debian9 stretch (which is currently not in the build map) there is no php5-cli until we integrate the sury repo with auto usage of it.

broken tests output:
```
Notice: /Stage[main]/Php::Repo::Debian/Apt::Key[php::repo::debian]/Apt_key[php::repo::debian]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]/ensure: defined content as '{md5}5a480294344d27cef9c39f8a50be08ea'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]: Scheduling refresh of Class[Apt::Update]
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]/ensure: defined content as '{md5}f117d407b06bee024f38f77dd7884d71'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]: Scheduling refresh of Class[Apt::Update]
Info: Class[Apt::Update]: Scheduling refresh of Exec[apt_update]
Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Php::Packages/Package[php5-cli]/ensure: created
```

broken sources output:
```
Notice: /Stage[main]/Php::Repo::Debian/Apt::Key[php::repo::debian]/Apt_key[php::repo::debian]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Package[apt-transport-https]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]/ensure: defined content as '{md5}388f39972254556224b8aeb890c1e672'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]: Scheduling refresh of Class[Apt::Update]
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]/ensure: defined content as '{md5}f097cbe4a1ea1fe281b93efd805303b3'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]: Scheduling refresh of Class[Apt::Update]
Info: Class[Apt::Update]: Scheduling refresh of Exec[apt_update]
Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
```

it could be a problem of the order of the source lists and that `wheezy` release is forced.

### PR DESCRIPTION
**Pull request by SimonHoenscheid**
…is tested. fixes #433

### CODE REVIEW

**Comment by bastelfreak** [comment]
I am not sure if this fixes the real issue, The acceptance tests worked in the past on debian 8, and now they don't. This just hides the hides but doesn't fix it.

**Comment by SimonHoenscheid** [comment]
What would you look for? A test for debian 7 running on a debian 8 box will never run successful. I dont get the point.

**Comment by bastelfreak** [comment]
okay, so dotdeb provides 5.6 for debian 8, which doesn't work. but even using it is a stupid idea since it's the default version shipped by debian? should we add a warning if somebody tries that, or add a note to the readme?

**Comment by bastelfreak** [comment]
@SimonHoenscheid I never wrote it as a debian 7 specific test, but I now think it worked only by accident on debian 8.

**Comment by SimonHoenscheid** [comment]
I would suggest removing dotdeb. According to this statement https://www.dotdeb.org/2017/01/27/php-7-1-dotdeb/

Wheezy LTS is EOL this month. This deprecates 5.6 for Wheezy. Due to the fact Sury offers 5.6, 7.0, 7.1, 7.2 for jessie and stretch so I would suggest only offering the Sury repo. With the hint it is not needed on stretch for 7.0 if people do not need multiple versions in parallel.

Surys packages allow multiple versions in parallel. The module should have this possibility too.

**Comment by bastelfreak** [comment]
@SimonHoenscheid sounds good! Can you provide a PR for that?

File: .travis.yml [hunk 1]
```diff
@@ -102,4 +102,4 @@ deploy:
     # all_branches is required to use tags
     all_branches: true
     # Only publish the build marked with "DEPLOY_TO_FORGE"
-    condition: "$DEPLOY_TO_FORGE = yes"
+    condition: "$DEPLOY_TO_FORGE = yes"
```
Comment (bastelfreak) [inline .travis.yml]: in general we want trailing newlines (documented at https://voxpupuli.org/docs/#reviewing-a-module-pr )

Comment (SimonHoenscheid) [reply .travis.yml]: Is there a chance to trigger a travis ci run, without changing the PR? I just removed the line to trigger a rebuild.

I move the dtailed discussion to a new issue

**Comment by SimonHoenscheid** [comment]
This PR can be closed, the changes where also part of the debian 9 change, to make the build green.

File: .travis.yml (hunk 1 above)
Comment (bastelfreak) [reply .travis.yml]: you can always rerun travis by closing and reopening a PR

Comment (SimonHoenscheid) [reply .travis.yml]: Thanks

**Pull request closed without merging**